docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+import io
+import os
+from shlex import split
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+from sys import stdin
+from typing import (Any, List, Optional, Sequence, Type, TypeVar, Union,
+                    overload)
+
+from .exceptions import CommandNotFoundOnImageError
+
+
+class Image:
+    """A class which holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
+
+    -   Capable of building docker images from dockerfiles or dockerfile-
+        formatted strings via the build classmethods.
+    -   Capable of acquiring information about the given image (see property
+        methods below).
+    -   Capable of running commands on containers built from the image using
+        the run method.
+    -   All docker build and run methods on the class can recive a file-like
+        object, (io.TextIOBase or some subclass of it,) and they will write the
+        outputs from docker into the file to enable logging.
+
+    NOTE: This class does not work correctly if the image built does not have
+        bash installed.
+    """
+    Self = TypeVar("Self", bound="Image")
+
+    def __init__(self, name_or_id: str):
+        """Initialize this Image. Connects the class with its ID.
```

Comment:
>`Connects the class with its ID.`

Note sure I understand what you mean by this?

I usually just write something generic for the `__init__()` description, like this:

```suggestion
        """Initialize a new Image object.
```